### ISSUE TITLE
"requires" installer option not working properly in dev branch

### ISSUE THREAD

**Issue by akien-mga** [lutris/lutris#172]
After discussing it with @Xodetaetl and doing some debugging myself, it looks like the "requires" installer option is not working properly in the current master and next branches.

I'm writing an installer for Warcraft III: The Frozen Throne which depends on having Warcraft III: Reign of Chaos installed. The latter is provided via: https://lutris.net/games/install/warcraft-iii-reign-of-chaos-cd-with-opengl.yml

For TFT, I use something similar:
```
description: CD or download version.
exe: drive_c/Program Files/Warcraft III/Frozen Throne.exe
files:
- {setup: 'N/A:Select the installer for Warcraft III: The Frozen Throne'}
game: {prefix: $GAMEDIR}
game_slug: warcraft-iii-the-frozen-throne
installer:
- task: {description: 'Installing Warcraft III: The Frozen Throne', executable: setup,
    name: wineexec, prefix: $GAMEDIR}
- move: {dst: $GAMEDIR/drive_c/Program Files/Warcraft III/Movies_, src: $GAMEDIR/drive_c/Program
      Files/Warcraft III/Movies/*}
name: 'Warcraft III: The Frozen Throne'
notes: Ingame videos are disabled (by adding an underscore to the movies folder name)
  to avoid crashing.
runner: wine
slug: warcraft-iii-the-frozen-throne-cd-with-opengl
requires: warcraft-iii-reign-of-chaos
steamid: null
version: CD (with OpenGL)
year: 2003
```

I tried using both the "slug" and the "game_slug" of Warcraft III: Reign of Chaos, but none will work with the latest master and next branches. I just get told that I need to install ${game_,}slug even though I already have it installed.

I tried with lutris 0.3.6.3, and there the "requires" parameter works fine when using the target's "game_slug".

**Comment by akien-mga** [comment]
Actually I'm not sure about the next branch, since I don't manage to install Warcraft III: Reign of Chaos at all, I guess the installer script is no longer compatible with next. I get:
```
ERROR:lutris.util.log:Error while completing task <bound method ScriptInterpreter.task of <lutris.installer.ScriptInterpreter object at 0x7f7d2806ae10>>: argument of type 'NoneType' is not iterable
```

**Comment by akien-mga** [comment]
I confirm the feature is not working as expected in the master branch, I just installed Age of Empires II: The Age of Kings, but I can't use the patch installer or install the expansion because it complains that "age-of-empires-ii-the-age-of-kings" is not installed.

**Comment by akien-mga** [comment]
I guess I identified the bogus commit: a7950e25cf
The "requires" feature now checks for the "installer_slug" field, but there is no installer_slug in the installer scripts that I checked.

Most scripts have only "slug" and "game_slug". Is "slug" meant to be the installer slug, and if so should it be renamed in existing scripts?

**Comment by akien-mga** [comment]
I checked that substituting "installer_slug" with "slug" in installer.py does fix this issue with existing scripts.

**Comment by Xodetaetl** [comment]
> I checked that substituting "installer_slug" with "slug" in installer.py does fix this issue with existing scripts.

Yes, but `installer_slug` is what we want to use from now on with `requires`. The problem is that the installer slug hasn't been recorded in the local database for games installed up until now. It's a bug, and I've pushed a fix for this on the website's repo, but we won't be able to recover it retroactively. Which means that games installed before the fix is deployed will have to be reinstalled if one wants to install a patch / mod / expansion for it in the future. I'm trying to think of ways to lessen the inconvenience...

Another issue we have to fix is that currently the game and installer slugs of the 'parent' game are overwritten when we install something with `requires`. 
What we need to do is check if something is installed by comparing with the installer slug field instead of the game slug. And keep track of the parent installer if there's one (add a `parent_installer` field).

**Comment by Xodetaetl** [comment]
Fixed: https://github.com/lutris/website/commit/b3c3d7678076e02b56d812a202b7d853dc40408c
